fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2675,8 +2675,13 @@ public CommandProcessingResult chargeOff(JsonCommand command) {
             loan.markAsChargedOff(transactionDate, currentUser, null);
         }
 
+        final List<Long> existingTransactionIds = loan.findExistingTransactionIds();
+        final List<Long> existingReversedTransactionIds = loan.findExistingReversedTransactionIds();
+
         LoanTransaction chargeOffTransaction = LoanTransaction.chargeOff(loan, transactionDate, txnExternalId);
+        loan.addLoanTransaction(chargeOffTransaction);
         loanTransactionRepository.saveAndFlush(chargeOffTransaction);
```

Comment:
since you are adding to loan and save the loan, you can remove the "loanTransactionRepository.saveAndFlush(chargeOffTransaction);"

Reply:
Fixed